src/main/java/org/openrewrite/kotlin/internal/KotlinTreeParserVisitor.java

```diff
@@ -3357,25 +3338,146 @@ private J.If.Else buildIfElsePart(KtIfExpression expression) {
     /*====================================================================
      * Type related methods
      * ====================================================================*/
-    private <T extends J> T mapType(T tree) {
-        return mapType(tree, null);
-    }
-
     @SuppressWarnings("unchecked")
-    private <T extends J> T mapType(T tree, @Nullable JavaType type) {
+    private <T extends J> T mapType(T tree) {
```

Comment:
@traceyyoshima Can you briefly explain why we need / want to do it this way rather than trying to set the appropriate type already at construction time? I think we should then also add a comment here to that extent, because to me this isn't entirely clear.

Reply:
Since this method is really just one switch and we often at the call aite already know the object's type, I would have overloaded the method for every type. I assume that we won't always know the type at compile time at the call site and will thus still require this method to delegate to the others.

I think we should however also think about a solution where we can vonstruct the correct type attribution to begin with. Possibly by doing something similar as we did for the generic type arguments in the element associations. Alternatively, if we don't think this is feasible, we should just add a brief comment here explaining why this fixup is required here.

Reply:
Ah, that does sound clearer -- I'll add overloads and comments about the changes.

AFAIK, there is not an alternative yet, mostly because we don't know what the mis-mappings are. The changes are meant to allow us to detect type mapping issues so that we can fix them proactively.